herddb-core/src/test/java/herddb/codec/RecordSerializerTest.java

```diff
@@ -79,4 +80,52 @@ private static void testTimestamp(String testCase, String timezone, long expecte
         assertEquals("failed for " + testCase + " delta is " + delta + " h, result is " + formattedResult, expectedResult, result.getTime());
     }
 
+    @Test
+    public void testSerializeWithNullAndNonNullTypes() {
+        byte iBytes [] = RecordSerializer.serialize(new Integer(10), ColumnTypes.INTEGER);
+        byte iBytesNonNullType[] = RecordSerializer.serialize(new Integer(10), ColumnTypes.NOTNULL_INTEGER);
+        assertArrayEquals(iBytes, iBytesNonNullType);
+
+        byte lBytes [] = RecordSerializer.serialize(new Long(1982), ColumnTypes.NOTNULL_LONG);
+        byte lBytesNonNullType[] = RecordSerializer.serialize(new Long(1982), ColumnTypes.LONG);
+        assertArrayEquals(lBytes, lBytesNonNullType);
+
+        byte sBytes [] = RecordSerializer.serialize("test", ColumnTypes.STRING);
+        byte sBytesNonNullType[] = RecordSerializer.serialize("test", ColumnTypes.NOTNULL_STRING);
+        assertArrayEquals(sBytes, sBytesNonNullType);
+    }
+
+    @Test
+    public void testSerializeThrowsExceptionOnNullObject() {
+        assertNull(RecordSerializer.serialize(null, ColumnTypes.STRING));
+    }
+
+    @Test
+    public void testDeserializeWithNullAndNonNullTypes() {
+        byte iValueAsByteArray[] = ByteBuffer.allocate(4).putInt(1000).array();
```

Comment:
you have to use h.u.Bytes.from_int, other wise you won't handle endianness consistently

don't use ByteBuffer, but always h.u.Bytes, even in the lines below